### PR TITLE
Scapy terminal improvements

### PR DESCRIPTION
**Pull request by gpotter2**
This commit changes the following:
- `.scapy_startup.py` and `.scapy_prestart.py` are moved to `.config/scapy/startup.py` and `.config/scapy/prestart.py`. This is what apparently most big libraries/softwares are moving towards (matplotlib, wireshark, etc.). ~~Question: do we want to move the history there? (for now it's the only file remaining in \~)~~ (done)
- automatically create `.config/scapy/prestart.py`. This is IMHO much easier for a novice user to understand what's going on, since it will show what it is supposed to look like.
- the Scapy terminal now supports **`ptpython`** and **`bpython`**, in addition to the existing **`IPython`**. `ptpython` comes with the **`ptipython`** variant that's a mix between `ptpython` and `IPython`.
- improve doc a bit, remove useless info `usage.rst`.
- fix https://github.com/secdev/scapy/issues/4056

Bugs:
- bpython requires the https://github.com/bpython/bpython/pull/992 and https://github.com/bpython/bpython/pull/996 patches to autocomplete classes similarly as the others
- ptpython requires https://github.com/prompt-toolkit/ptpython/pull/545 and https://github.com/prompt-toolkit/ptpython/issues/542. The default `repr` is also broken

I encourage you to try them all a bit, some are pretty fun. Until one of them is as stable as IPython though, it will remain the default. There are some `conf.color_theme` features that comes with IPython that are not implemented here yet.

#### Gallery

**bpython**
image

**ptpython**
image

### CODE REVIEW

**Comment by guedou** [comment]
@gpotter2 I am fine with moving everything into `.config/scapy` that makes more sense.

File: scapy/main.py [hunk 1]
```diff
@@ -412,11 +449,29 @@ def update_session(fname=None):
     update_ipython_session(scapy_session)
 
 
+@overload
```
Comment (guedou) [inline scapy/main.py:452]: Can you explain the rationale to overload `init_session()`?  I did not understand it on my own.

File: scapy/main.py [hunk 2]
```diff
@@ -659,8 +701,110 @@ def interact(mydict=None, argv=None, mybanner=None, loglevel=logging.INFO):
...
+
+    # Check IPython import
+    if conf.interactive_shell in ["ipython", "ptipython"]:
+        try:
+            import IPython  # noqa: F811
+        except ImportError:
+            log_loading.warning("IPython requested but not found !")
+            conf.interactive_shell = "python"
+
+    # Check bpython import
```
Comment (guedou) [inline scapy/main.py:749]: Could factor the Python shell checks? The following lines looks redundant.

File: scapy/main.py [hunk 3]
```diff
@@ -694,18 +838,54 @@ def interact(mydict=None, argv=None, mybanner=None, loglevel=logging.INFO):
...
+                _kwargs["configure"] = ptpython_configure
             try:
-                start_ipython(config=cfg, user_ns=SESSION)
+                embed(config=cfg, user_ns=SESSION, **_kwargs)
             except (AttributeError, TypeError):
                 code.interact(banner=banner_text, local=SESSION)
-    else:
+    # Start ptpython
+    elif conf.interactive_shell == "ptpython":
+        log_loading.warning("ptpython support is currently partially broken")
```
Comment (guedou) [inline scapy/main.py:858]: Can you add a comment explaining why?

**Review by guedou — CHANGES_REQUESTED** [review]

File: scapy/main.py (hunk 3 above)
Comment (gpotter2) [reply scapy/main.py:850]: Done !

File: scapy/main.py (hunk 1 above)
Comment (guedou) [reply scapy/main.py:455]: Thanks!

**Comment by gpotter2** [comment]
@guedou @polybassa @p-l- Ready for review & merge !

**Comment by guedou** [comment]
LGTM.Thanks for the changes. I don't why but I cannot approve the PR.

**Review by polybassa — APPROVED** [review]

**Review by p-l- — APPROVED** [review]

**Pull request merged**